prime-router/src/testIntegration/kotlin/datatests/mappinginventory/dtm/DTMTests.kt

```diff
@@ -0,0 +1,73 @@
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/instant/dtm-csec-precision-to-instant").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/instant/dtm-msec-precision-to-instant").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/instant/dtm-mmsec-precision-to-instant").passed)
+    }
+
+    @Test
+    fun `test zero dtm to instant`() {
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/instant/dtm-zero-year-to-instant").passed)
+    }
+
+    @Test
+    fun `test dtm to dateTime`() {
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/dateTime/dtm-year-precision-to-dateTime").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/dateTime/dtm-month-precision-to-dateTime").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/dateTime/dtm-day-precision-to-dateTime").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/dateTime/dtm-minute-precision-to-dateTime").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/dateTime/dtm-second-precision-to-dateTime").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/dateTime/dtm-dsec-precision-to-dateTime").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/dateTime/dtm-csec-precision-to-dateTime").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/dateTime/dtm-msec-precision-to-dateTime").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/dateTime/dtm-mmsec-precision-to-dateTime").passed)
+    }
+
+    @Test
+    fun `test zero dtm to dateTime`() {
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/dateTime/dtm-zero-year-to-dateTime").passed)
+    }
+
+    @Test
+    fun `test dtm to date`() {
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/date/dtm-year-precision-to-date").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/date/dtm-month-precision-to-date").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/date/dtm-day-precision-to-date").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/date/dtm-minute-precision-to-date").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/date/dtm-second-precision-to-date").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/date/dtm-dsec-precision-to-date").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/date/dtm-csec-precision-to-date").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/date/dtm-msec-precision-to-date").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/date/dtm-mmsec-precision-to-date").passed)
+    }
+
+    @Test
+    fun `test zero dtm to date`() {
+        assert(verifyHL7ToFHIRToHL7Mapping("dtm/date/dtm-zero-year-to-date").passed)
+    }
+
+    @Test
+    fun `test dt to dateTime`() {
+        assert(verifyHL7ToFHIRToHL7Mapping("dt/dt-year-precision-to-dateTime").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dt/dt-month-precision-to-dateTime").passed)
+        assert(verifyHL7ToFHIRToHL7Mapping("dt/dt-day-precision-to-dateTime").passed)
+    }
+
+//    @Test
+//    fun `test zero dt to dateTime`() {
```

Comment:
@arnejduranovic Do you think this case supporting `0000` in NK1.8/9 is necessary? If we don't have to support values that register as non-valid dates, we don't have to use an extension for DT<->dateTime, but we can also just use the extension to make this work without too much trouble.

Reply:
We talked about this over slack, and it sounds like there could be reason to support this. Will try to uncomment this test and update the NK1.8/9 mapping to support this case.